### PR TITLE
feat: broadcast frame to cascading zippers

### PR DESCRIPTION
**Pull request by venjiang**
broadcast frame to cascading zippers

- - -
This change is [Reviewable](https://reviewable.io/reviews/yomorun/yomo/366)

### CODE REVIEW

File: core/frame/data_frame.go [hunk 1]
```diff
@@ -70,6 +70,16 @@ func (d *DataFrame) SourceID() string {
 	return d.metaFrame.SourceID()
 }
 
+// SetDispatch set dispatch mode
+func (d *DataFrame) SetDispatch(mode Dispatch) {
```
Comment (wujunzhuo) [inline core/frame/data_frame.go:74]: in DataFrame, `SetBroadcast` might be better

Comment (venjiang) [reply core/frame/data_frame.go:74]: I'm not sure we have any other model.

**Comment by fanweixiao** [comment]
*[`core/frame/meta_frame.go` line 17 at r1](https://reviewable.io/reviews/yomorun/yomo/366#-N9kDZHg7b8OYyKUwgyZ:-N9kDZHg7b8OYyKUwgy_:b-cu3wtt) ([raw file](https://github.com/yomorun/yomo/blob/3923830511a447542e5bbf020b7560e21e838292/core/frame/meta_frame.go#L17)):*
> ```Go
> 	metadata []byte
> 	sourceID string
> 	dispatch Dispatch
> ```

Is `boolean` better than `Dispatch`?

**Comment by fanweixiao** [comment]
*[`core/frame/data_frame_test.go` line 22 at r1](https://reviewable.io/reviews/yomorun/yomo/366#-N9kBTTv6Rdv3OQbFw9t:-N9kBTTv6Rdv3OQbFw9u:b-n1lwmm) ([raw file](https://github.com/yomorun/yomo/blob/3923830511a447542e5bbf020b7560e21e838292/core/frame/data_frame_test.go#L22)):*
> ```Go
> 	result = append(result, tidbuf...)
> 	result = append(result, byte(TagOfSourceID), 0x0)
> 	result = append(result, byte(TagOfDispatch), 0x1, 0x1)
> ```

`TagOfDispatch` only present when value is `true`.



_Code quote (from source.go):_
```Go
f.SetDispatch(frame.DispatchBroadcast)
```

**Comment by fanweixiao** [comment]
*[`core/frame/data_frame.go` line 79 at r1](https://reviewable.io/reviews/yomorun/yomo/366#-N9kEPTX0VxBrR3XN86r:-N9kEPTX0VxBrR3XN86s:bx1wugv) ([raw file](https://github.com/yomorun/yomo/blob/3923830511a447542e5bbf020b7560e21e838292/core/frame/data_frame.go#L79)):*
> ```Go
> 
> // Dispatch get dispatch mode
> func (d *DataFrame) Dispatch() Dispatch {
> ```

should be `IsDispatch()` better?

**Comment by fanweixiao** [comment]
Shall we uniform the word, `dispatch` or `broadcast`?

**Review by fanweixiao — APPROVED** [review]
> Reviewed 8 of 8 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/yomorun/yomo/366)* status: all files reviewed, 4 unresolved discussions (waiting on @venjiang and @wujunzhuo)

**Comment by venjiang** [comment]
> Shall we uniform the word, `dispatch` or `broadcast`?

Yeah, It should be named broadcast if there is only one distribution mode.
Can we have other dispatch modes?

**Review by wujunzhuo — APPROVED** [review]

**Pull request merged**